hook-consumer/src/consumer.rs

```diff
@@ -174,7 +175,11 @@ async fn send_webhook(
         .headers(headers)
         .body(body)
         .send()
-        .await?;
+        .await
+        .map_err(|e| WebhookConsumerError::RetryableWebhookError {
+            reason: e.to_string(),
```

Comment:
Since we surface Webhook send errors to users, I think we'll want to tame this with an enum rather than leaving it open ended, but that seems unrelated to this change and I'm leaving it for later.